### PR TITLE
Bugfix/resolve liveness test bug

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

**Review by LukasGasior1 — APPROVED** [review]
LGTM

**Review by siy — APPROVED** [review]
LGTM

File: radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/FProposalsPerViewDropper.java [hunk 1]
```diff
@@ -59,12 +59,17 @@ public boolean test(MessageInTransit msg) {
 			final View view = proposal.getVertex().getView();
 			final Set<BFTNode> nodesToDrop = proposalToDrop.computeIfAbsent(view, v -> {
 				List<BFTNode> nodes = new LinkedList<>(validatorSet);
-				Set<BFTNode> nextFaultySet = new HashSet<>();
-				for (int i = 0; i < faultySize; i++) {
-					BFTNode nextFaultyNode = nodes.remove(random == null ? i : random.nextInt(nodes.size()));
-					nextFaultySet.add(nextFaultyNode);
+				ImmutableSet.Builder<BFTNode> nextFaultySet = ImmutableSet.builder();
```
Comment (clstrfsck) [inline radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/FProposalsPerViewDropper.java:62]: Better here would be to shuffle `nodes` with `random`, and remove the first `faultySize` items.  Much less code and also not quadratic complexity.

File: radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/FProposalsPerViewDropper.java [hunk 2]
```diff
@@ -59,12 +59,17 @@ public boolean test(MessageInTransit msg) {
...
+				if (random != null) {
+					for (int i = 0; i < faultySize; i++) {
+						BFTNode nextFaultyNode = nodes.remove(random.nextInt(nodes.size()));
+						nextFaultySet.add(nextFaultyNode);
+					}
+				} else {
+					nodes.stream().limit(faultySize).forEach(nextFaultySet::add);
```
Comment (clstrfsck) [inline radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/FProposalsPerViewDropper.java:69]: Would be better as `nextFaultySet.addAll(nodes.subList(0, faultySize))`.  Could be shared with non-null case too, like this:
```
    final List<BFTNode> nodes = Lists.newArrayList(validatorSet);
    if (random != null) {
        Collections.shuffle(nodes, random);
    }
    final ImmutableSet<BFTNode> nextFaultySet = ImmutableSet.copyOf(nodes.subList(0, faultySize));
```

File: radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/SimulationNetwork.java [hunk 3]
```diff
@@ -128,7 +128,7 @@ public String toString() {
 	@Inject
 	public SimulationNetwork(ChannelCommunication channelCommunication) {
 		this.channelCommunication = Objects.requireNonNull(channelCommunication);
-		this.receivedMessages = ReplaySubject.<MessageInTransit>createWithSize(20) // To catch startup timing issues
+		this.receivedMessages = ReplaySubject.<MessageInTransit>createWithSize(128) // To catch startup timing issues
```
Comment (clstrfsck) [inline radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/SimulationNetwork.java:131]: Maybe this should be a function of the number of nodes in the system?

Also, nice catch.  Must have been a bit of a pain to track down.

**Review by clstrfsck — APPROVED** [review]
Some minor suggestions.

File: radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/FProposalsPerViewDropper.java [hunk 4]
```diff
@@ -59,12 +59,17 @@ public boolean test(MessageInTransit msg) {
 			final View view = proposal.getVertex().getView();
 			final Set<BFTNode> nodesToDrop = proposalToDrop.computeIfAbsent(view, v -> {
 				List<BFTNode> nodes = new LinkedList<>(validatorSet);
-				Set<BFTNode> nextFaultySet = new HashSet<>();
-				for (int i = 0; i < faultySize; i++) {
-					BFTNode nextFaultyNode = nodes.remove(random == null ? i : random.nextInt(nodes.size()));
```
Comment (clstrfsck) [inline radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/FProposalsPerViewDropper.java:64]: Slightly nasty bug there.  Why did this not trigger some sort of out of range exception?

I guess we are using `faultySize <<< validatorSet.size()`.

**Pull request closed without merging**